WalletWasabi/Affiliation/CoinjoinRequestsUpdater.cs

```diff
@@ -0,0 +1,216 @@
+using System.Linq;
+using System.Collections.Immutable;
+using System.Collections.Generic;
+using WalletWasabi.Affiliation.Models;
+using NBitcoin;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.WabiSabi.Backend.Events;
+using WalletWasabi.Bases;
+using System.Collections.Concurrent;
+using WalletWasabi.Affiliation.Models.CoinjoinRequest;
+
+namespace WalletWasabi.Affiliation;
+
+public class CoinjoinRequestsUpdater : PeriodicRunner
+{
+	private static readonly TimeSpan AffiliateServerTimeout = TimeSpan.FromSeconds(60);
+	private static readonly TimeSpan Interval = TimeSpan.FromSeconds(1);
```

Comment:
I don't see any problem with this as it is, or am I misunderstanding it @kiminuo? I mean, there are no 86400 HTTP requests a day, are they?

Reply:
> I would expect an async queue[1](https://github.com/zkSNACKs/WalletWasabi/pull/9769#user-content-fn-1-165384a995b6596bb2900d516246fa50) that waits until it has an element in it. Or maybe that channel that Lucas mentioned can be used in the similar way.

Done in https://github.com/zkSNACKs/WalletWasabi/pull/9769.

An alternative is to use
  * [AsyncQueue](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.threading.asyncqueue-1) from `Microsoft.VisualStudio.Threading` or
  * [AsyncProducerConsumerQueue](https://github.com/StephenCleary/AsyncEx/blob/master/src/Nito.AsyncEx.Coordination/AsyncProducerConsumerQueue.cs) from `Nito.AsyncEx`
 
as you mentioned. Since both alternatives require a new dependency and implementation of asynchronous queue using [channels](https://learn.microsoft.com/en-us/dotnet/core/extensions/channels) seems pretty straightforward to me, I dediced to implement it myself.